### PR TITLE
Clarify how to run client-side code in framework components

### PR DESCRIPTION
**Pull request by caseywatts**
#### What kind of changes does this PR include?
- New or updated content

#### Description

- Clarifies how to run client-side code within framework components - they require a lifecycle method AND a `client:` directive (this has taken me days to figure out! lol glad to document it here)

### CODE REVIEW

**Comment by Jutanium** [comment]
Thanks for contributing to the docs! It's this line that you added that I think is super helpful for our readers:

> Tell the framework component to hydrate client-side by using a [`client:`](/en/reference/directives-reference/) directive.

That said, we like the way our existing section is structured. We don't use nested bullet points in Astro docs, and "entirely cient-side rendered" vs "partially client-side rendered" isn't quite how it works - in both cases, the component is completely client-side rendered; in the case of `client:` attributes other than `client:only`, it's _also_ server-side rendered.

Would you be willing to add the note about using a `client:` directive to our existing paragraph setup?

**Comment by Jutanium** [comment]
Hey @caseywatts, are you still interested in this one? Happy to take it over.

**Comment by sarah11918** [comment]
Hey @Jutanium -- I think it's ok if you jump in with this one now! Let's get @caseywatts' contribution into the docs!  :partying_face:

File: src/content/docs/en/guides/troubleshooting.mdx [hunk 1]
```diff
@@ -31,9 +31,18 @@ Framework components run on the server by default, so this error can occur when
 
 **Solution**: Determine the code that calls `document` or `window`. If you aren't using `document` or `window` directly and still getting this error, check to see if any packages you're importing are meant to run on the client. 
 
-- If the code is in an Astro component, move it to a `<script>` tag outside of the frontmatter. This tells Astro to run this code on the client, where `document` and `window` are available.
+If the code is in an Astro component
```
Comment (Jutanium) [inline src/content/docs/en/guides/troubleshooting.mdx:34]:
```suggestion
- If the code is in an Astro component, move it to a `<script>` tag outside of the frontmatter. This tells Astro to run this code on the client, where `document` and `window` are available.

- If the code is in a framework component, try to access these objects after rendering using lifecycle methods (e.g. [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) in React, [`onMounted()`](https://vuejs.org/api/composition-api-lifecycle.html#onmounted) in Vue, and [`onMount()`](https://svelte.dev/docs#run-time-svelte-onmount) in Svelte). Tell the framework component to hydrate client-side by using a [client:](https://github.com/en/reference/directives-reference/) directive, like `client:load`, to run these lifecycle methods. You can also prevent the component from rendering on the server at all by adding the [`client:only`](/en/reference/directives-reference/#clientonly) directive.
```

File: src/content/docs/en/guides/troubleshooting.mdx [hunk 2]
```diff
@@ -33,7 +33,18 @@ Framework components run on the server by default, so this error can occur when
...
+- move it to a `<script>` tag outside of the frontmatter.
+  - This tells Astro to run this code on the client, where `document` and `window` are available.
+
+If the code is in a framework component, there are two ways:
+
+- Entirly client-side rendered
+  - You can prevent the component from rendering on the server at all by adding the [`client:only`](/en/reference/directives-reference/#clientonly) directive.
+- Partially client-side rendered, which requires two steps:
+  - Tell the framework component to hydrate client-side by using a [`client:`](/en/reference/directives-reference/) directive.
+  - Within the framework component, use a lifecycle method to specify which parts should be client-side only. e.g. [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) in React, [`onMounted()`](https://vuejs.org/api/composition-api-lifecycle.html#onmounted) in Vue, and [`onMount()`](https://svelte.dev/docs#run-time-svelte-onmount) in Svelte.
```
Comment (Jutanium) [inline src/content/docs/en/guides/troubleshooting.mdx:47]:
```suggestion

```

**Pull request merged**